game/sample_game/nakama/private_key.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+// private_key.go manages the creation and loading of the Nakama private key used to sign
+// all transactions.
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/heroiclabs/nakama-common/runtime"
+)
+
+var (
+	ErrorNoStorageObjectFound       = errors.New("no storage object found")
+	ErrorTooManyStorageObjectsFound = errors.New("too many storage objects found")
+
+	globalPrivateKey    *ecdsa.PrivateKey
+	globalSignerAddress string
+	// nonceMutex is used to ensure that Nakama's current nonce can be read and incremented atomically.
+	nonceMutex sync.Mutex
+)
+
+const (
+	privateKeyCollection = "private_key_collection"
+	privateKeyKey        = "private_key_key"
+	privateKeyNonce      = "private_key_nonce"
+	adminAccountID       = "00000000-0000-0000-0000-000000000000"
+)
+
+type privateKeyStorageObj struct {
+	Value string
+}
+
+// getOnePKStorageObj loads one specific runtime.StorageObject from the privateKeyCollection in
+// Nakama's storage layer. An error is returned if too few or too many storage objects are found.
+func getOnePKStorageObj(ctx context.Context, nk runtime.NakamaModule, key string) (string, error) {
+	objs, err := nk.StorageRead(ctx, []*runtime.StorageRead{{
+		Collection: privateKeyCollection,
+		UserID:     adminAccountID,
+		Key:        key,
+	}})
+	if err != nil {
+		return "", err
+	}
+	if len(objs) > 1 {
+		return "", ErrorTooManyStorageObjectsFound
+	} else if len(objs) == 0 {
+		return "", ErrorNoStorageObjectFound
+	}
+	var pkObj privateKeyStorageObj
+	if err := json.Unmarshal([]byte(objs[0].Value), &pkObj); err != nil {
+		return "", err
+	}
+	return pkObj.Value, nil
+}
+
+// setOnePKStorageObj saves the given value to the privateKeyCollection in Nakama's storage layer.
+func setOnePKStorageObj(ctx context.Context, nk runtime.NakamaModule, key, value string) error {
```

Comment:
can we rename `key` and `value` to reflect what we actually need to pass into the function?